src/include/daos/dpar.h

```diff
@@ -31,7 +31,7 @@ par_version_compatible(uint32_t version)
 	if (major != DPAR_MAJOR) /* Total incompatibility */
 		return false;
 
-	if (minor < DPAR_MINOR) /* An API we use doesn't exist in other library */
+	if (minor > DPAR_MINOR) /* An API used by the caller doesn't exist in this library */
```

Comment:
Sorry, not quite understand the logic, why not check "minor == DPAR_MINOR" or "minor != DPAR_MINOR"? although the result is the same as "minor > DPAR_MINOR" since DPAR_MINOR is zero here.

According to the commit comment, as long as the library contains the expected API, then it is NOT "incompatible". But the caller init_routine() checks the version compatibility when load symbol successfully, does it conflict with this comment?

static void
init_routine(void)
{
        bool            fail = false;
        uint32_t        version;

        stubs_handle = dlopen("libdpar_mpi.so", RTLD_NOW);

        if (stubs_handle == NULL) {
                printf("No MPI found, using serial library\n");
                return;
        }

        FOREACH_PAR_SYMBOL(LOAD_SYM, fail);

        if (!fail) {
                version = stubs.ps_getversion();
                if (par_version_compatible(version)) {
                        printf("Using compatible version\n");
                        return;
                }

Reply:
If the caller has a lower minor version, it means that the APIs it is using are all implemented.  If it has a higher minor version, it means it is probably using an API that doesn't exist because it was compiled against a version that added new APIs.

Reply:
Minor version bump means new APIs were added. Major version bump means APIs were changed or removed in an non-binary compatible way.

Reply:
Hmm, this is in the header file so it's probably wrong.  Let me upload a new patch.